plugin/src/main/java/org/wildfly/plugins/bootablejar/maven/common/LegacyPatchCleaner.java

```diff
@@ -0,0 +1,254 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.plugins.bootablejar.maven.common;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+import org.apache.maven.plugin.logging.Log;
+import org.jboss.galleon.util.IoUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+/**
+ *
+ * @author jdenise
+ */
+public class LegacyPatchCleaner {
+
+    private final Path wildflyDir;
+    private final Set<Path> existingModules;
+    private final Log log;
+    private final Path overlays;
+    private final Path modulesDir;
+    private final Path overlaysFile;
+    private final Path patchesDir;
+    public LegacyPatchCleaner(Path wildflyDir, Log log) throws IOException {
+        this.wildflyDir = wildflyDir;
+        modulesDir = wildflyDir.resolve("modules").resolve("system").resolve("layers").resolve("base");
+        overlays = modulesDir.resolve(".overlays");
+        overlaysFile = overlays.resolve(".overlays");
+        patchesDir = wildflyDir.resolve(".installation").resolve("patches");
+        this.log = log;
+        existingModules = captureModules();
+    }
+
+    private Log getLog() {
+        return log;
+    }
+
+    private Set<Path> captureModules() throws IOException {
+        Set<Path> existingModules = new HashSet<>();
+        Files.walkFileTree(modulesDir, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path t, BasicFileAttributes bfa) throws IOException {
+                if (t.getFileName().toString().equals("module.xml")) {
+                    existingModules.add(modulesDir.relativize(t).getParent());
+                }
+                return FileVisitResult.CONTINUE;
+            }
+        });
+        return existingModules;
+    }
+
+    public void clean() throws Exception {
+        cleanupModules();
+    }
+
+    private void cleanupModules() throws Exception {
+
+        getLog().info("Legacy patch cleanup enabled, checking for unused resources...");
+
+        if (!Files.exists(overlays)) {
```

Comment:
This is not a big deal at all just a readability thing I prefer, but again doesn't matter :) You could use `Files.notExists` so it's not a negative test.

Reply:
Thank-you, fixed.